File: src/main/java/com/freya02/botcommands/api/components/event/StringSelectEvent.java
package com.freya02.botcommands.api.components.event;

import com.freya02.botcommands.api.BContext;
import com.freya02.botcommands.internal.BContextImpl;
import net.dv8tion.jda.api.events.interaction.component.StringSelectInteractionEvent;

public class StringSelectEvent extends StringSelectInteractionEvent {
	private final BContext context;

	public StringSelectEvent(BContextImpl context, StringSelectInteractionEvent event) {
		super(event.getJDA(), event.getResponseNumber(), event.getInteraction());

		this.context = context;
	}

	public BContext getContext() {
		return context;
	}
}

File: src/main/kotlin/com/freya02/botcommands/api/components/event/StringSelectEvent.kt
package com.freya02.botcommands.api.components.event

import com.freya02.botcommands.api.BContext
import com.freya02.botcommands.api.commands.ratelimit.CancellableRateLimit
import net.dv8tion.jda.api.events.interaction.component.StringSelectInteractionEvent

class StringSelectEvent internal constructor(
    val context: BContext,
    event: StringSelectInteractionEvent,
    cancellableRateLimit: CancellableRateLimit
) : StringSelectInteractionEvent(event.jda, event.responseNumber, event.interaction),
    CancellableRateLimit by cancellableRateLimit